src/components/ChatEntry.js

```diff
@@ -1,22 +1,44 @@
 import React from 'react';
 import './ChatEntry.css';
 import PropTypes from 'prop-types';
+import TimeStamp from './TimeStamp';
+
+const ChatEntry = ({ onUpdate, ...message }) => {
+  const onLikeButtonClick = () =>
+    onUpdate({
+      ...message,
+      liked: !message.liked,
+    });
+
+  const [likedClass, likedIcon] = message.liked
+    ? ['liked', '❤️']
+    : ['unliked', '🤍'];
+  const localRemoteClass = message.localRemote ? 'local' : 'remote';
 
-const ChatEntry = (props) => {
   return (
-    <div className="chat-entry local">
-      <h2 className="entry-name">Replace with name of sender</h2>
+    <div className={`chat-entry ${localRemoteClass}`}>
+      <h2 className="entry-name">{message.sender}</h2>
       <section className="entry-bubble">
-        <p>Replace with body of ChatEntry</p>
-        <p className="entry-time">Replace with TimeStamp component</p>
-        <button className="like">🤍</button>
+        <p>{message.body}</p>
+        <p className="entry-time">
+          <TimeStamp time={message.timeStamp}></TimeStamp>
+        </p>
+        <button className={`like ${likedClass}`} onClick={onLikeButtonClick}>
+          {likedIcon}
+        </button>
       </section>
     </div>
   );
 };
 
 ChatEntry.propTypes = {
-  //Fill with correct proptypes
+  id: PropTypes.number.isRequired,
+  sender: PropTypes.string.isRequired,
+  body: PropTypes.string.isRequired,
+  timeStamp: PropTypes.string.isRequired,
+  liked: PropTypes.bool.isRequired,
+  localRemote: PropTypes.bool,
+  onUpdate: PropTypes.func,
```

Comment:
I would probably make both of these `.isRequired`, but considering that then breaks the tests due to the lacking test data we instructors wrote, fair enough leaving them not required.

Reply:
This also applies to `onupdate`, `localUser`, and `updateChatEntry` in `ChatLog`, but I decided to add to this comment instead of making a new one.